doc/reference/reference_sql/sql_user_guide.rst

```diff
@@ -808,10 +895,44 @@ value IS [NOT] NULL |br|
 CASE ... WHEN ... THEN ... ELSE ... END |br|
   ... for setting a series of conditions.
 
+.. _sql_map_expression:
+
+{ key : value } |br|
+... for MAP expressions. |br|
+Literal Examples: ``{'a':1}``, ``{ "column_1" : X'1234' }`` |br|
+Non-literal Examples: ``{"a":"a"}``, ``{UUID(), (SELECT 1) + 1}`` |br|
+An expression has data type = MAP if it is enclosed in curly brackets
+(also called braces) ``{`` and ``}`` and contains a key for identification,
+then a colon ``:``, then a value for what the key identifies.
+The key data type must be INTEGER or STRING or UUID.
+The value data type may be anything.
+The Lua equivalent type is 'map' but the syntax is slightly different,
+for example the SQL value ``{'a': 1}`` is represented in Lua as ``{a = 1}``.
+
+.. _sql_array_expression:
+
+[ value ... ] |br|
+... for ARRAY expressions. |br|
+Examples: ``[1,2,3,4]``, ``[1,[2,3],4]``, ``['a', "column_1", uuid()]`` |br|
+An expression has data type = ARRAY if it is a sequence of zero or more values
+enclosed in square brackets (``[`` and ``]``).
+Often the values in the sequence are called "elements".
+The element data type may be anything, including ARRAY -- that is, ARRAYs may be nested.
+Different elements may have different types.
+The Lua equivalent type is `'array' <https://www.lua.org/pil/11.1.html>`_.
+
+.. _sql_array_index_expression:
+
+ARRAY index expression: |br|
```

Comment:
I think we should expand this part to include `[]` for MAP. Or it should be dropped, since the patch is currently in review: https://github.com/tarantool/tarantool/pull/7140